discovery-provider/src/tasks/index_rewards_manager_backfill.py

```diff
@@ -0,0 +1,648 @@
+            "slot": result["slot"],
+            "timestamp": result["blockTime"],
+            "transfer_instruction": None,
+        }
+        meta = result["meta"]
+        if meta["err"]:
+            logger.info(
+                f"index_rewards_manager_backfill.py | Skipping error transaction from chain {tx_info}"
+            )
+            return tx_metadata
+        tx_message = result["transaction"]["message"]
+        instruction = get_valid_instruction(tx_message, meta)
+        if instruction is None:
+            return tx_metadata
+        transfer_instruction_data = parse_transfer_instruction_data(instruction["data"])
+        amount = transfer_instruction_data["amount"]
+        eth_recipient = transfer_instruction_data["eth_recipient"]
+        id = transfer_instruction_data["id"]
+        transfer_instruction = parse_transfer_instruction_id(id)
+        if transfer_instruction is None:
+            return tx_metadata
+
+        challenge_id, specifier = transfer_instruction
+        receiver_index = instruction["accounts"][TRANSFER_RECEIVER_ACCOUNT_INDEX]
+        pre_balance, post_balance = get_solana_tx_token_balances(meta, receiver_index)
+        if pre_balance == -1 or post_balance == -1:
+            raise Exception("Reward recipient balance missing!")
+        tx_metadata["transfer_instruction"] = {
+            "amount": amount,
+            "eth_recipient": eth_recipient,
+            "challenge_id": challenge_id,
+            "specifier": specifier,
+            "prebalance": pre_balance,
+            "postbalance": post_balance,
+        }
+        return tx_metadata
+    except Exception as e:
+        logger.error(
+            f"index_rewards_manager_backfill.py | Error processing {tx_sig}, {e}",
+            exc_info=True,
+        )
+        raise e
+
+
+def process_batch_sol_reward_manager_txs(
+    session: Session,
+    reward_manager_txs: List[RewardManagerTransactionInfo],
+):
+    """Validates that the transfer instruction is consistent with DB and inserts ChallengeDisbursement DB entries"""
+    try:
+        logger.info(f"index_rewards_manager | processing {reward_manager_txs}")
+        eth_recipients = [
+            tx["transfer_instruction"]["eth_recipient"]
+            for tx in reward_manager_txs
+            if tx["transfer_instruction"] is not None
+        ]
+        users = (
+            session.query(User.wallet, User.user_id, UserBankAccount.bank_account)
+            .join(UserBankAccount, UserBankAccount.ethereum_address == User.wallet)
+            .filter(User.wallet.in_(eth_recipients), User.is_current == True)
+            .all()
+        )
+
+        users_map = {
+            user[0]: {"user_id": user[1], "user_bank": user[2]} for user in users
+        }
+
+        specifiers = [
+            tx["transfer_instruction"]["specifier"]
+            for tx in reward_manager_txs
+            if tx["transfer_instruction"] is not None
+        ]
+        user_challenges = (
```

Comment:
I think we need to be careful here not to re-create these old user_challenges

Reply:
I think we're just querying them here, think I removed the code that inserts into the challenges db.